src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/writers/JSONLDOutputHelper.java

```diff
@@ -0,0 +1,182 @@
+/* (c) 2021 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.featurestemplating.writers;
+
+import static org.geoserver.featurestemplating.builders.EncodingHints.CONTEXT;
+import static org.geoserver.featurestemplating.builders.VendorOptions.COLLECTION_NAME;
+import static org.geoserver.featurestemplating.builders.VendorOptions.JSONLD_TYPE;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Logger;
+import org.geoserver.catalog.Catalog;
+import org.geoserver.catalog.FeatureTypeInfo;
+import org.geoserver.featurestemplating.builders.EncodingHints;
+import org.geoserver.featurestemplating.builders.VendorOptions;
+import org.geoserver.featurestemplating.builders.impl.RootBuilder;
+import org.geoserver.featurestemplating.builders.impl.TemplateBuilderContext;
+import org.geoserver.featurestemplating.configuration.TemplateIdentifier;
+import org.geoserver.featurestemplating.configuration.TemplateLoader;
+import org.geoserver.featurestemplating.utils.FeatureTypeInfoUtils;
+import org.geoserver.ows.Dispatcher;
+import org.geoserver.ows.Request;
+import org.geoserver.platform.GeoServerExtensions;
+import org.geotools.feature.FeatureCollection;
+import org.geotools.feature.FeatureIterator;
+import org.geotools.util.logging.Logging;
+import org.opengis.feature.Feature;
+
+/** This class provides methods to handle a JSON-LD output writing outside of WFS Response class. */
+public class JSONLDOutputHelper {
```

Comment:
Does it need to be public?

Reply:
The JSONLDFeatureResponse class use the hints related method too, so to avoid code repetition I've made the class public.
Anyway since the write method is used only in this package I switched it to package default.